cylc/flow/network/schema.py

```diff
@@ -1182,4 +1183,104 @@ class Mutations(ObjectType):
         description=TaskActions._meta.description)
 
 
-schema = Schema(query=Queries, mutation=Mutations)
+# ** Subscription Related ** #
+
+def to_subscription(func: Callable, sleep_seconds: float = 5.) -> Callable:
+    """Wraps a function in a while-true-sleep, transforming
+    the function into an async-generator, used by the
+    websockets/subscriptions.
+
+    Args:
+        func (Callable): a callable.
+        sleep_seconds (float): asyncio sleep interval in seconds.
+    Returns:
+        Callable: a callable async-generator wrapping the original callable.
+    """
+    async def gen(*args: Any, **kwargs: Any) -> AsyncGenerator[Any, None]:
+        """
+        Args:
+            *args: Variable length argument list, varies as per schema.
+            **kwargs: Arbitrary keyword arguments, varies as per schema.
+        Returns:
+            AsyncGenerator[Any, None]: an async generator that will yield values from
+                resolvers.
+        """
+        while True:
+            yield await func(*args, **kwargs)
+            await asyncio.sleep(sleep_seconds)
+    return gen
+
+
+class Subscriptions(ObjectType):
+    """Defines the subscriptions available in the schema."""
+    class Meta:
+        description = """Multi-Workflow root level subscriptions."""
+    workflows = List(
+        Workflow,
+        description=Workflow._meta.description,
+        ids=List(ID, default_value=[]),
+        exids=List(ID, default_value=[]),
+        resolver=to_subscription(get_workflows))
```

Comment:
@dwsutherland now I can query 

`subscription { workflows }`

Or 

`subscription { tasks }`

However, querying

`subscription { workflows { tasks }  }`

Doesn't bring me anything. Any clue what I could be missing?

Reply:
Will have a look

Reply:
Oh, never mind. Just realized why that's happening. In the `class Workflow(ObjectType)`, it has the `Tasks` and the resolver is set there too. Hmmm, tricky to get this one working for subscriptions now :disappointed:

Reply:
I think someone had the same issue with the aiohttp version:
https://github.com/graphql-python/graphql-ws/issues/12#issuecomment-476989150

Looks as though they modified the handler to await all async nested fields..

Reply:
:eyes: gonna read it again tomorrow. That method could be a life saver. Just need to understand the logic and see if that'd work for us too. Great find @dwsutherland !

Reply:
@dwsutherland , quick test, and that actually worked. Will update the PR (just clean up current mess, move customizations to a separate file, etc) in a few minutes. Understood about 50% of the code, but will re-read it in the morning :coffee: Thanks **a ton** for that. Otherwise I would still be scratching my head with no solution.